src/components/EventRegistration.js

```diff
@@ -0,0 +1,103 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import RegistrationCollection from './RegistrationCollection';
+
+class EventRegistration extends Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      collections: this.props.formGroups.reduce((acc, reg) => {
+        acc[reg.id] = [];
+        return acc;
+      }, {})
+    };
+    this.updateCollection = this.updateCollection.bind(this);
+    this.submit = this.submit.bind(this);
+    this.mergeDataAndFormGroups = this.mergeDataAndFormGroups.bind(this);
+  }
+
+  /*
+   * Is called when parent updates props, i.e. when data is updated from the outside
+   */
+  componentWillReceiveProps(nextProps) {
+    if (nextProps && nextProps.data) {
+      this.setState({ collections: this.mergeDataAndFormGroups(nextProps.data) });
+    }
+  }
+
+  mergeDataAndFormGroups(data) {
+    const result = {...data};
+    // add formGroups that are not yet present in data
+    this.props.formGroups.forEach((fg, index) => {
+      if (result[fg.id] === undefined) {
+        result[fg.id] = [];
+      }
+    });
+    return result;
+  }
+
+  collect = (data, id) => {
+    this.setState({
+      collect: {
+        ...this.state.collect,
+        [id]: data,
+      },
+    });
+  };
```

Comment:
tip: når du skal oppdatere state og trenger den forrige staten, er det bedre å gjøre slik:
`
this.setState( (prevState) => {
 return {
    collect: {...prevState, [id]: data }
  }
})
`

Les mer om det her: https://facebook.github.io/react/docs/react-component.html#setstate